Assets/Plugins/Source/Editor/Windows/CreatePackageWindow.cs

```diff
@@ -203,22 +212,41 @@ protected override void RenderWindow()
                     labelSize.x,
                     labelSize.y);
 
-                EditorGUI.ProgressBar(progressBarRect, _progress, "");
+                lock (_progressLock)
+                {
+                    _displayedProgress = Mathf.Lerp(_displayedProgress, _actualProgress, Time.deltaTime * 2);
+                }
+
+                EditorGUI.ProgressBar(progressBarRect, _displayedProgress, "");
 
                 GUI.Label(labelRect, _progressText, customLabelStyle);
 
                 GUILayout.Space(20f);
                 if (GUILayout.Button("Cancel"))
                 {
-                    FileUtility.CleanDirectory(_packagingConfig.pathToOutput);
-                    _progress = 0.0f;
+                    _actualProgress = 0.0f;
                     _progressText = "";
                     _createPackageCancellationTokenSource?.Cancel();
+                    FileUtility.CleanDirectory(_packagingConfig.pathToOutput);
                 }
                 GUILayout.EndVertical();
             }
         }
 
+        private void UpdateProgressBar(CancellationToken cancellationToken)
+        {
+            const int smoothingFactor = 500;
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                Thread.Sleep(smoothingFactor);
+
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
This makes me feel like something else was going to go on in here. while cancellation isn't requested, sleep, then check if cancellation is requested, if not then loop.

Reply:
I think is tied to the fact that the function name it is within is also misleading. But I'll address those concerns in a separate PR where I move all of the ProgressBar stuff into its own class, so that `CreatePackageWindow` does not have to include code for ProgressBar - something it really is out of scope for it to know.